c10/core/TensorImpl.cpp

```diff
@@ -104,24 +104,16 @@ TensorImpl::TensorImpl(
 // the Python and PythonTLSSnapshot dispatch keys will be set and all is well.
 // The point is to delay the dispatch key setting until that point.
 
-// NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
 TensorImpl::TensorImpl(
     ImplType type,
     Storage&& storage,
     DispatchKeySet key_set,
     const caffe2::TypeMeta data_type)
-    : storage_(std::move(storage)),
-
-      numel_(0),
-      data_type_(data_type),
-      device_opt_(storage_.device()),
-      key_set_(key_set - c10::python_ks) { // See [Note: Python key removal]
-  init_bitfields();
-  // Inference tensor doesn't have version counter.
-  if (!is_inference()) {
-    version_counter_ = VariableVersion(/*version=*/0);
-  }
-}
+    : TensorImpl(
+          std::move(storage),
+          key_set,
+          data_type,
+          /*shadow_storage=*/nullptr) {}
```

Comment:
I think you need to do the subclasses too. Once you've expunged all the old VIEW sites you should delete this constructor entirely, as it is a footgun that makes it easy to incorrectly propagate shadow storage.